src/signature.rs

```diff
@@ -144,7 +144,12 @@ impl <'a> SignedRequest <'a> {
     /// Calculate the signature from the credentials provided and the request data
     /// Add the calculated signature to the request headers and execute it
     /// Return the hyper HTTP response
-    pub fn sign_and_execute(&mut self, creds: AwsCredentials) -> Response {
+    pub fn sign_and_execute(&mut self, creds: &AwsCredentials) -> Response {
```

Comment:
This breaks all the codegen, as you can see from the Travis build.  The generated code isn't expecting to pass a reference.

This is by design, because the generated clients don't store credentials to which they'd want to pass a reference. They keep a `ProvideAwsCredentials` which is asked fresh credentials each time they're needed, since it's possible that the credentials have changed or expired.  The various `ProvideAwsCredentials` implementations are responsible for caching/refreshing as needed.